notifications.yaml

```diff
@@ -0,0 +1,268 @@
+                Location (URL) of accepted message resource.
+
+                A GET request to this URL returns the message status and
+                details.
+definitions:
+  ProblemJson:
+    title: Problem Type
+    type: object
+    properties:
+      type:
+        type: string
+        format: uri
+        description: |
+          An absolute URI that identifies the problem type.  When dereferenced,
+          it SHOULD provide human-readable documentation for the problem type
+          (e.g., using HTML).
+        default: 'about:blank'
+        example: 'https://example.com/problem/constraint-violation'
+      title:
+        type: string
+        description: |
+          A short, summary of the problem type. Written in english and readable
+          for engineers (usually not suited for non technical stakeholders and
+          not localized); example: Service Unavailable
+      status:
+        type: integer
+        format: int32
+        description: >
+          The HTTP status code generated by the origin server for this
+          occurrence
+
+          of the problem.
+        minimum: 100
+        maximum: 600
+        exclusiveMaximum: true
+        example: 503
+      detail:
+        type: string
+        description: |
+          A human readable explanation specific to this occurrence of the
+          problem.
+        example: Connection to database timed out
+      instance:
+        type: string
+        format: uri
+        description: >-
+          An absolute URI that identifies the specific occurrence of the
+          problem.
+
+          It may or may not yield further information if dereferenced.
+  MessageContent:
+    title: MessageContent
+    type: object
+    properties:
+      body_short:
+        type: string
+        description: Message content (short version).
+        minLength: 3
+        maxLength: 100
+      body_long:
+        type: string
+        description: Message content (long version).
+        minLength: 3
+    required:
+      - body_short
+  MessageDefinition:
+    title: MessageDefinition
+    type: object
+    properties:
+      to:
+        description: The fiscal code of the destination user.
+        type: string
+      dry_run:
+        type: boolean
+        description: >-
+          This parameter, when set to true, allows developers to test a request
+          without actually sending a message.
+
+
+          The default value is false.
+      time_to_live:
+        type: integer
+        description: >-
+          This parameter specifies how long (in seconds) the message should be
+          kept in storage if the device is offline. The maximum time to live
+          supported is 1 year, and the default value is 1 hour.
+        minimum: 3600
+        maximum: 31536000
+      content:
+        $ref: '#/definitions/MessageContent'
+    required:
+      - to
+      - content
+  MessageStatus:
+    title: MessageStatus
+    type: object
+    properties:
```

Comment:
probably you're right and we can collapse status. however this exposes new issues we never considered before ie:

> why the sender, that cannot decide what channels the message should be sent to, may want to know the status of delivery for each channel ?

because when she sends `body_short` and `body_long` (et al.) wants to know which part of the payload has been delivered (ie. if you just return the ) and not all channels support all payload types. (in this case probably it's better to have a status for each payload type and abstract the channel away :)

> but just tell the sender whether the message as been delivered to at least one channel

> how is the information that some channel succeeded but some other channel failed useful to the sender?

same as above. you lose the info on which part of the payload gets delivered.